core/src/main/java/org/codice/imaging/nitf/core/common/NitfDateTime.java

```diff
@@ -80,10 +82,13 @@ public final String getSourceString() {
         This is a best-effort, based on the information available, which can be incomplete. If
         no other information is available, a default Date will be returned.
 
-        @return the value of this object as a Date.
-    */
+        @return the value of this object as a Date, or null if the date is not valid.
+     */
```

Comment:
Ok, can we just wrap the source string and a ZoneDateTime in one class?  That should give clients everything they need to convert the date in any way they need.